### ISSUE TITLE
libnvcuvid.so is missing

### ISSUE THREAD

**Issue by carlwain74** [NVIDIA/nvidia-docker#1001]
I am trying to build a docker image based off CUDA 10.1 for Ubuntu 16.04.
The application running in the docker container requires libnvcuvid.so. However, this library is not present in the Docker container.

In order to for the application to run successfully I have had to copy the libnvcuvid.so.418.67 library from host machine (which has the CUDA 10.1 toolkit installed) to the /usr/lib/x86_64-linux-gnu/ folder.

I have defined the following environment settings in my Dockerfile

ENV NVIDIA_VISIBLE_DEVICES all
ENV NVIDIA_DRIVER_CAPABILITIES video,compute,utility

Is there a more elegant way to obtain this library (i.e. via an apt package)

 - [x] Kernel version from `uname -a`
Linux 5ab363ee00c7 4.15.0-50-generic #54~16.04.1-Ubuntu SMP Wed May 8 15:55:19 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
 - [x] Driver information from `nvidia-smi -a`
```
root@5ab363ee00c7:/opt/RealNetworks/virgo# nvidia-smi -a

==============NVSMI LOG==============

Timestamp                           : Tue Jun 25 17:14:36 2019
Driver Version                      : 418.67
CUDA Version                        : 10.1

Attached GPUs                       : 1
GPU 00000000:41:00.0
    Product Name                    : GeForce GTX 1080 Ti
    Product Brand                   : GeForce
    Display Mode                    : Enabled
    Display Active                  : Enabled
    Persistence Mode                : Disabled
    Accounting Mode                 : Disabled
    Accounting Mode Buffer Size     : 4000
    Driver Model
        Current                     : N/A
        Pending                     : N/A
    Serial Number                   : N/A
    GPU UUID                        : GPU-1641dd98-e9ea-8d91-8f58-60d1cf7122d4
    Minor Number                    : 0
    VBIOS Version                   : 86.02.39.00.90
    MultiGPU Board                  : No
    Board ID                        : 0x4100
    GPU Part Number                 : N/A
    Inforom Version
        Image Version               : G001.0000.01.04
        OEM Object                  : 1.1
        ECC Object                  : N/A
        Power Management Object     : N/A
    GPU Operation Mode
        Current                     : N/A
        Pending                     : N/A
    GPU Virtualization Mode
        Virtualization mode         : None
    IBMNPU
        Relaxed Ordering Mode       : N/A
    PCI
        Bus                         : 0x41
        Device                      : 0x00
        Domain                      : 0x0000
        Device Id                   : 0x1B0610DE
        Bus Id                      : 00000000:41:00.0
        Sub System Id               : 0x65913842
        GPU Link Info
            PCIe Generation
                Max                 : 3
                Current             : 3
            Link Width
                Max                 : 16x
                Current             : 16x
        Bridge Chip
            Type                    : N/A
            Firmware                : N/A
        Replays Since Reset         : 0
        Replay Number Rollovers     : 0
        Tx Throughput               : 0 KB/s
        Rx Throughput               : 0 KB/s
    Fan Speed                       : 20 %
    Performance State               : P2
    Clocks Throttle Reasons
        Idle                        : Not Active
        Applications Clocks Setting : Not Active
        SW Power Cap                : Not Active
        HW Slowdown                 : Not Active
            HW Thermal Slowdown     : Not Active
            HW Power Brake Slowdown : Not Active
        Sync Boost                  : Not Active
        SW Thermal Slowdown         : Not Active
        Display Clock Setting       : Not Active
    FB Memory Usage
        Total                       : 11177 MiB
        Used                        : 3716 MiB
        Free                        : 7461 MiB
    BAR1 Memory Usage
        Total                       : 256 MiB
        Used                        : 5 MiB
        Free                        : 251 MiB
    Compute Mode                    : Default
    Utilization
        Gpu                         : 4 %
        Memory                      : 1 %
        Encoder                     : 0 %
        Decoder                     : 7 %
    Encoder Stats
        Active Sessions             : 0
        Average FPS                 : 0
        Average Latency             : 0
    FBC Stats
        Active Sessions             : 0
        Average FPS                 : 0
        Average Latency             : 0
    Ecc Mode
        Current                     : N/A
        Pending                     : N/A
    ECC Errors
        Volatile
            Single Bit
                Device Memory       : N/A
                Register File       : N/A
                L1 Cache            : N/A
                L2 Cache            : N/A
                Texture Memory      : N/A
                Texture Shared      : N/A
                CBU                 : N/A
                Total               : N/A
            Double Bit
                Device Memory       : N/A
                Register File       : N/A
                L1 Cache            : N/A
                L2 Cache            : N/A
                Texture Memory      : N/A
                Texture Shared      : N/A
                CBU                 : N/A
                Total               : N/A
        Aggregate
            Single Bit
                Device Memory       : N/A
                Register File       : N/A
                L1 Cache            : N/A
                L2 Cache            : N/A
                Texture Memory      : N/A
                Texture Shared      : N/A
                CBU                 : N/A
                Total               : N/A
            Double Bit
                Device Memory       : N/A
                Register File       : N/A
                L1 Cache            : N/A
                L2 Cache            : N/A
                Texture Memory      : N/A
                Texture Shared      : N/A
                CBU                 : N/A
                Total               : N/A
    Retired Pages
        Single Bit ECC              : N/A
        Double Bit ECC              : N/A
        Pending                     : N/A
    Temperature
        GPU Current Temp            : 57 C
        GPU Shutdown Temp           : 96 C
        GPU Slowdown Temp           : 93 C
        GPU Max Operating Temp      : N/A
        Memory Current Temp         : N/A
        Memory Max Operating Temp   : N/A
    Power Readings
        Power Management            : Supported
        Power Draw                  : 60.10 W
        Power Limit                 : 250.00 W
        Default Power Limit         : 250.00 W
        Enforced Power Limit        : 250.00 W
        Min Power Limit             : 125.00 W
        Max Power Limit             : 300.00 W
    Clocks
        Graphics                    : 1480 MHz
        SM                          : 1480 MHz
        Memory                      : 5005 MHz
        Video                       : 1265 MHz
    Applications Clocks
        Graphics                    : N/A
        Memory                      : N/A
    Default Applications Clocks
        Graphics                    : N/A
        Memory                      : N/A
    Max Clocks
        Graphics                    : 1911 MHz
        SM                          : 1911 MHz
        Memory                      : 5505 MHz
        Video                       : 1620 MHz
    Max Customer Boost Clocks
        Graphics                    : N/A
    Clock Policy
        Auto Boost                  : N/A
        Auto Boost Default          : N/A
    Processes
        Process ID                  : 4596
            Type                    : G
            Name                    : /usr/lib/xorg/Xorg
            Used GPU Memory         : 29 MiB
        Process ID                  : 16009
            Type                    : C
            Name                    : /opt/RealNetworks/virgo/virgofeedd
            Used GPU Memory         : 3674 MiB
```
 - [x] Docker version from `docker version`
Docker version is 18.09.6
 - [x] NVIDIA packages version from `dpkg -l '*nvidia*'` _or_ `rpm -qa '*nvidia*'`
No packages found
 - [x] Docker command, image and tag used
FROM nvidia/cuda:10.1-base-ubuntu16.04

**Comment by RenaudWasTaken** [comment]
You should check if libnvcuvid is picked up by your host's ldconfig (ldconfig-p | grep cuvid).
With the following command, my container get libnvcuvid mounted from the host:
```
docker run -it --runtime nvidia -e NVIDIA_DRIVER_CAPABILITIES=video,compute,utility nvidia/cuda:10.0-runtime-ubuntu16.04 sh -c 'ldconfig -p | grep cuvid'
	libnvcuvid.so.1 (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libnvcuvid.so.1
```

**Comment by carlwain74** [comment]
So this issue has been resolved. Initially we were deploying both CUDA toolkit and drivers. Now we are deploying just the drivers (latest).
Also when we deploy our system we remove existing docker installations and deploy the expected release. During this process the runtime was getting deleted as a dependency of docker. Once we corrected the install of nvidia-docker2 after docker is installed things cleared up. We can see that the appropriate libraries are getting mounted into the container and our application is running as expected
